app/src/main/java/com/rkm/tasky/util/date/DateUtils.kt

```diff
@@ -1,15 +1,36 @@
 package com.rkm.tasky.util.date
 
 import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
 import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
 import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
 import kotlinx.datetime.TimeZone
 import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.format.MonthNames
+import kotlinx.datetime.format.Padding
+import kotlinx.datetime.format.char
+import kotlinx.datetime.minus
 import kotlinx.datetime.toInstant
 import kotlinx.datetime.toLocalDateTime
 import kotlinx.datetime.todayIn
 
 private val timezone = TimeZone.currentSystemDefault()
+private val timeFormat = LocalTime.Format {
+    amPmHour(padding = Padding.ZERO)
+    char(':')
+    minute(padding = Padding.ZERO)
+    char(' ')
+    amPmMarker(am = "AM", pm = "PM")
+}
+private val dateFormat = LocalDate.Format {
+    monthName(names = MonthNames.ENGLISH_ABBREVIATED)
+    char(' ')
+    dayOfMonth(padding = Padding.ZERO)
+    char(' ')
+    year(Padding.NONE)
+}
```

Comment:
This is a bit verbose and there are built-in formats for the outputs you’ve created:

```
private val timeFormat = DateTimeFormatter.ofPattern("hh:mm a") // e.g., 09:45 AM
private val dateFormat = DateTimeFormatter.ofPattern("MMM dd yyyy") // e.g., Nov 14 2024
```

Reply:
I'm using a kotlin date time library. That why it looks a little funky. lol. I would use DateTimeFormatter though.